lib/collection/src/collection/mod.rs

```diff
@@ -509,6 +530,20 @@ impl Collection {
     }
 
     pub async fn remove_shards_at_peer(&self, peer_id: PeerId) -> CollectionResult<()> {
+        let resharding_state = self
+            .resharding_state()
+            .await
+            .filter(|state| state.peer_id == peer_id);
+
+        if let Some(state) = resharding_state {
+            if let Err(err) = self.abort_resharding(state.key(), true).await {
+                log::error!(
+                    "Failed to abort resharding {} while removing peer {peer_id}: {err}",
+                    state.key(),
+                );
+            }
+        }
```

Comment:
I was a bit confused by this one. A simple comment might help.

```suggestion
        // If shards are removed from peer driving resharding, abort it now
        let resharding_state = self
            .resharding_state()
            .await
            .filter(|state| state.peer_id == peer_id);

        if let Some(state) = resharding_state {
            if let Err(err) = self.abort_resharding(state.key(), true).await {
                log::error!(
                    "Failed to abort resharding {} while removing peer {peer_id}: {err}",
                    state.key(),
                );
            }
        }
```

Reply:
Sure, but it's not "shards removed from peer", but "the peer".

We can't remove last replica of a shard, so we can't remove the one-and-only new shard (at least, not until it was replicated).